### DISCUSSION TITLE
Github runner support for Windows and Linux<title>

### DISCUSSION THREAD

**Discussion by vhuang-cnx** [community/community#44919] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

We are trying to build our C++ sources on 2 OSs but have questions if we can use GitHub's runners.
For the Windows, we need to use Visual C++ 2015 compiler, does GitHub's Windows runner support/have it?
For the Linux, we are running build on CentOS 7 and GitHub has only ubuntu based on doc. Is there a CentOS 7 runner?
Or the only option for us is to run self hosted runner?

Thanks.

**Comment by moncefdhk999** [comment]
je ne peux pas faire un organisme sans tous je suis bloque

**Reply by moncefdhk999** [reply]
je ne peux pas faire un organisme sans vous meme je suis bloque

**Comment by airtower-luna** [comment]
You can find the descriptions of the GitHub runners here, including lists of installed software: https://github.com/actions/runner-images#available-images (Click the version you want in the table.)

> For the Linux, we are running build on CentOS 7 and GitHub has only ubuntu based on doc. Is there a CentOS 7 runner?

You can use [job containers](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontainer) with a suitable image. That way your job runs on an Ubuntu runner in a container based on the image of your choice.

For CentOS specifically note that the [official Docker image](https://hub.docker.com/_/centos) as well as [CentOS as a whole](https://www.centos.org/centos-linux-eol/) are deprecated. You should still be able to do something like this:

```yaml
    runs-on: ubuntu-latest
    container:
      image: centos:7
```

But in the long run you should probably look for a different distro for your build. :warning:

**Reply by vhuang-cnx** [reply]
Thanks for your quick reply. 

We did check the installed software but only saw VS 2019 installed:

Visual Studio Enterprise 2019 | 16.11.33214.272 | C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise

which does not have VC++ 2015. Not sure if missed anything?

**Reply by airtower-luna** [reply]
If it's not listed it probably isn't installed. You could try to install it during the workflow, but I don't use Windows, so I can't say how that'd work or if there'd be license issues.

**Reply by vhuang-cnx** [reply]
Thank you very much
